File: 1160. Find Words That Can Be Formed by Characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        st = collections.Counter(chars)
        cs = [collections.Counter(s) for s in words]
        return sum([len(s) for i,s in enumerate(words) if all(st[c]>=cs[i][c] for c in cs[i])])

File: 1160. Find Words That Can Be Formed by Characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        vector<int> cnt(26);
        for (auto c : chars) ++cnt[c-'a'];
        int ans = 0;
        for (auto& s : words) {
            vector<int> a(26);
            for (auto c : s) ++a[c-'a'];
            bool ok = true;
            for (int i=0; i<26; ++i)
                if (a[i] > cnt[i]) {
                    ok = false;
                    break;
                }
            if (ok) ans += s.size();
        }
        return ans;
    }
};